core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -66,6 +67,12 @@ public static EsThreadPoolExecutor newScaling(String name, int min, int max, lon
         return executor;
     }
 
+    public static EsThreadPoolExecutor newBoundedCached(String name, int poolSize, long keepAliveTime, TimeUnit unit, ThreadFactory threadFactory, ThreadContext contextHolder) {
+        EsThreadPoolExecutor executor = new EsThreadPoolExecutor(name, poolSize, poolSize, keepAliveTime, unit, new LinkedBlockingQueue<>(), threadFactory, new EsAbortPolicy(), contextHolder);
```

Comment:
> also we can't use the EsAbortPolicy - we may never reject. We should use ForceQueuePolicy

@bleskes I think that we should use a caller runs policy. The only case in which the rejected execution policy matters is if the queue fills up. But the queue has capacity of `Integer.MAX_VALUE`; if we ever hit that we are in deep trouble and a force queue policy will not help (the queue is full, no amount of forcing is going to get the task in the queue). Thus, caller runs.